tests/test-sessions/test-mongodb-sessions/src/test/java/org/eclipse/jetty/nosql/mongodb/MongoTestHelper.java

```diff
@@ -226,13 +225,13 @@ public static void createUnreadableSession (String id, String contextPath, Strin
 
         if (attributes != null)
         {
-            for (String name : attributes.keySet())
-            {
-                Object value = attributes.get(name);
-                sets.put(MongoSessionDataStore.__CONTEXT + "." + vhost.replace('.', '_') + ":" + contextPath+ "." + MongoUtils.encodeName(name),
-                         MongoUtils.encodeName(value));
-            }
+            SessionData tmp = new SessionData (id, contextPath, vhost, created, accessed, lastAccessed, maxIdle, attributes);
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ObjectOutputStream oos = new ObjectOutputStream(baos);
+            SessionData.serializeAttributes(tmp, oos);
```

Comment:
`ObjectOutputStream` not closed, use try-with-resources.

Reply:
ok